Fixes broken link to browserslist website, adds refernce to vault page. (#255)
Shopify/web-configs: packages/browserslist-config/README.md

| Safari                      | >=10    |
| iOS                         | >=10    |

You can list all supported browsers by running `npx browserslist "last 1 firefoxandroid versions, last 3 chrome versions, last 3 chromeandroid versions, last 3 firefox versions, last 3 opera versions, last 3 edge versions, safari >= 10, ios >= 10"`

> To learn more about browser and iOS support at Shopify visit [our vault page](https://vault.shopify.io/pages/1524-Browser-support)

Reviewer: ```suggestion
Shopify employees can learn more by visiting the [browser support Vault page](https://vault.shopify.io/pages/1524-Browser-support)
```
